pkg/cluster-runtime/apps.go

```diff
@@ -147,3 +148,30 @@ func (i AppInstaller) save(applicationFile string) error {
 
 	return nil
 }
+
+func GetImageDefaultLaunchCmds(extension v12.ImageExtension) []string {
+	rawCmds := extension.Launch.Cmds
+	appNames := extension.Launch.AppNames
+	// if raw cmds exist in extension, return it.
```

Comment:
In my opinion, the launch applications should take precedence over cmds, and we can even mark cmd  deprecated in the future because the app can completely override cmd's capabilities